src/v/raft/configuration_manager.cc

```diff
@@ -67,16 +67,35 @@ ss::future<> configuration_manager::prefix_truncate(model::offset offset) {
     vlog(_ctxlog.trace, "Prefix truncating configurations at {}", offset);
     return _lock.with([this, offset] {
         auto it = _configurations.lower_bound(offset);
+        /**
+         * When prefix truncation would remove all the configuration we instert
+         * the last configuration from before requested offset at the offset.
+         * This way we preserver last know configuration and indexing.
+         *
+         *                          200
+         *                           │
+         *                           │
+         *                           │ prefix truncate
+         *                           │
+         * ┌───────┬───────┐         ▼
+         * │   0   │  100  │...........................
+         * └───────┴───┬───┘
+         *             │              ┌───────┐
+         *             └─────────────►│  200  │...............
+         *                            └───────┘
+         *                   move last know configuration to truncate offset
+         *
+         *  NOTE: box with number represent an entry in configuration manager
+         */
+
         if (it == _configurations.end()) {
-            // we can not prefix truncate all configurations
-            return ss::make_exception_future<>(
-              std::invalid_argument(fmt::format(
-                "can not prefix truncate configuration manager at {} as last "
-                "available configuration has offset {}",
-                offset,
-                get_latest_offset())));
+            _configurations.emplace(offset, std::move(std::prev(it)->second));
```

Comment:
an assert that `std::prev(it)` is a valid iterator would be nice

Reply:
I've added following assertion:
```c++
        vassert(
          !_configurations.empty(),
          "Configuration manager should always have at least one "
          "configuration");
```
this way if `it==_configurations.end()` then `std::prev(it)` is always valid